counsel.el

```diff
@@ -4918,6 +4918,36 @@ An extra action allows to switch to the process buffer."
 ;; `counsel-slime-repl-history' within
 ;; `counsel--browse-history-action'.
 
+;;** `counsel-esh-dir-history'
+(declare-function eshell/cd "eshell")
+
+(defun counsel--esh-dir-history-action-cd (pair)
+  "Default action for counsel-esh-dir-history. It changes the
+current working directory to the one selected by the user."
+  (eshell/cd (car pair)))
+
+(defun counsel--esh-dir-history-action-edit (pair)
+  "Action for counsel-esh-dir-history to insert the selected
+directory to the Eshell buffer prefixed by \"cd \", allowing the
+caller to modify parts of the directory before switching to it."
+  (ivy--action-insert (format "cd %s" (car pair))))
+
+(defvar eshell-last-dir-ring)
+
+;;;###autoload
+(cl-defun counsel-esh-dir-history ()
+  "Use Ivy to navigate and jump through Eshell's directory stack."
```

Comment:
Not sure what navigating and jumping through mean in this context. Perhaps it would be simpler to say "complete" or "browse"?

Reply:
The idea behind using "jump" was to make clear that `counsel-esh-dir-history` allows not just to browse the stack but also to change the working directory by selecting a candidate. OTOH, I used "to navigate" as a replacement for "to browse". I've corrected the docstring to make it simpler following your suggestion :)

Reply:
Thanks, looks good.